### ISSUE TITLE
Temperature time sensors are not timestamp data type

### ISSUE THREAD

**Issue by jazzyisj** [Pirate-Weather/pirate-weather-ha#275]
### Describe the bug

The recently added time sensors for high and low temperatures are not timestamp data type.

### Expected behavior

A properly instantiated datetime entity can be used to display the date or time in a human friendly format in the UI like this using the [entity format parameter](https://www.home-assistant.io/dashboards/entities/#format).

```
  - entity: sensor.outdoor_high_temperature_time
    name: "High Temperature Time"
    format: time
```

<img width="360" alt="image" src="https://github.com/user-attachments/assets/c718fb69-07cd-48c1-883a-3c979cf9a075">

### Actual behavior

The Pirateweather time state sensors are not proper timestamps and do not work with the entity format parameter in the UI.

```
      - entity: sensor.pirateweather_daytime_high_apparent_temperature_time_0d
        name: "High Temperature Time" 
        format: time
```

<img width="371" alt="image" src="https://github.com/user-attachments/assets/80d3b17d-391a-48f6-a691-5ac2aa08ad46">

### Home Assistant version

2024.8.0b2

### Integration version

1.5.3

### Other details

https://developers.home-assistant.io/docs/core/entity/datetime/

### Troubleshooting steps

- [X] I have updated my Home Assistant installation to the latest version.
- [X] I have updated the Pirate Weather Integration to the latest version.
- [X] I have gone through the [documentation](https://pirateweather.net/en/latest/ha/) before opening this issue.
- [X] I have searched this repository and [API Repository](https://github.com/alexander0042/pirate-weather-ha) to see if the issue has already been reported.
- [X] I have restarted my Home Assistant installation.
- [X] I have queried the API in my browser to confirm the issue is not with the API.
- [X] I have written an informative title.

**Comment by cloneofghosts** [comment]
Thanks for opening this issue. What's happening here is those sensors in the API are in epoch time and not a timestamp. You'll need to convert them to a timestamp.

I'm not sure if its possible to convert them into a timestamp in the integration itself so I'll ping @alexander0042 to see if its possible

**Comment by alexander0042** [comment]
Good catch here, and sorry for the slow reply- I was working on fixing some stability issues with the primary API, so that was the focus. The good news here is that this was an easy change, and will be out in 1.5.5 today!

**Comment by cloneofghosts** [comment]
V1.5.5 has just been released which contains this fix.

**Comment by jazzyisj** [comment]
The format for  high temperature time sensors and apparent over night low have been updated to datetime but the forecast time and the daily low temperature time sensors have not.  Also, the daytime high temperature does not have the timezone in the timestamp where the apparent daytime high and overnight low do.


<img width="420" alt="image" src="https://github.com/user-attachments/assets/03620974-37d8-4a63-b209-b32acb66fe7d">
<img width="425" alt="image" src="https://github.com/user-attachments/assets/07f7a6ea-9476-4e11-a366-fbfea23bab2e">
<img width="422" alt="image" src="https://github.com/user-attachments/assets/b42676a6-99ea-4aa9-9603-3191e063ecb3">
<img width="416" alt="image" src="https://github.com/user-attachments/assets/0371fd03-fc85-4999-ae8f-e92c25a59728">
<img width="421" alt="image" src="https://github.com/user-attachments/assets/44cd7537-150f-4c45-9faa-40bcaca8443d">

**Comment by jazzyisj** [comment]
Hate to be a pain, but the device_class of the pirate weather time sensors is still not a timestamp. 

Here is a properly formatted timestamp sensor from the Environment Canada integration.

<img width="802" alt="image" src="https://github.com/user-attachments/assets/4234f510-dd2e-4078-b047-efe3dd781491">

This is a Pirateweather time sensor.

<img width="934" alt="image" src="https://github.com/user-attachments/assets/6f584438-b584-4cc0-a997-cb846cff909c">

The daily low temperature was missed altogher.

<img width="937" alt="image" src="https://github.com/user-attachments/assets/3386c31f-0767-4563-82fa-6fb5932a893c">

**Comment by alexander0042** [comment]
Thanks for mentioning the Environment Canada integration! I'll take a look at it to see what they do and go from there!

**Comment by cloneofghosts** [comment]
@alexander0042 Was testing the update in a devcontainer before creating a new release and am running into this error with the daily low sensor:

```
Logger: homeassistant.components.sensor
Source: helpers/entity_platform.py:598
integration: Sensor (documentation, issues)
First occurred: 10:28:22 AM (3 occurrences)
Last logged: 10:32:12 AM

Error adding entity sensor.pirateweather_daily_low_temperature_time_0d for domain sensor with platform pirateweather
Traceback (most recent call last):
  File "/home/vscode/.local/lib/python3.12/site-packages/homeassistant/components/sensor/__init__.py", line 594, in state
    if value.tzinfo is None:
       ^^^^^^^^^^^^
AttributeError: 'int' object has no attribute 'tzinfo'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/vscode/.local/lib/python3.12/site-packages/homeassistant/helpers/entity_platform.py", line 598, in _async_add_entities
    await coro
  File "/home/vscode/.local/lib/python3.12/site-packages/homeassistant/helpers/entity_platform.py", line 912, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/home/vscode/.local/lib/python3.12/site-packages/homeassistant/helpers/entity.py", line 1366, in add_to_platform_finish
    self.async_write_ha_state()
  File "/home/vscode/.local/lib/python3.12/site-packages/homeassistant/helpers/entity.py", line 1005, in async_write_ha_state
    self._async_write_ha_state()
  File "/home/vscode/.local/lib/python3.12/site-packages/homeassistant/helpers/entity.py", line 1130, in _async_write_ha_state
    self.__async_calculate_state()
  File "/home/vscode/.local/lib/python3.12/site-packages/homeassistant/helpers/entity.py", line 1067, in __async_calculate_state
    state = self._stringify_state(available)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.12/site-packages/homeassistant/helpers/entity.py", line 1011, in _stringify_state
    if (state := self.state) is None:
                 ^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.12/site-packages/homeassistant/components/sensor/__init__.py", line 605, in state
    raise ValueError(
ValueError: Invalid datetime: sensor.pirateweather_daily_low_temperature_time_0d has timestamp device class but provides state 1724230800:<class 'int'> resulting in ''int' object has no attribute 'tzinfo''
```

**Comment by alexander0042** [comment]
Thanks for testing this! I missed that variable since it has a different name than the others, but just pushed a commit that should fix it

**Comment by cloneofghosts** [comment]
Yup, that fixed the issue. One thing I noticed during testing was the Overnight Low Time sensor was showing the data from `temperatureMinTime` instead of `temperatureLowTime` so it shows the low as being 6h ago instead of being in 18h.

**Comment by cloneofghosts** [comment]
Other than the issue I pointed out above (maybe it's working as intended?) are we good to release a new version? Been holding off releasing a new version to make sure its stable. I've tested locally and everything works so I think we're good to release the changes?

**Comment by alexander0042** [comment]
Yea, I was a little confused by the high/low min/max for that one as well. I initially changed it to match to description in the text, so it's pulling the high/low (Day/Night) times for both apparent and temperature. However, in the entity description it's coded as min, and I think to keep things consistent it's better not to change it, so I just pushed a change to set it back to min

**Comment by cloneofghosts** [comment]
Tested the changes again and everything is looking good. Is this good to release or do you want to look at #241 first?

**Comment by alexander0042** [comment]
Let's leave that one open until we hear back if those sensors would be useful to have and push a new release in the meantime!

**Comment by cloneofghosts** [comment]
Alright will leave that one open. @jazzyisj I've released v1.5.8 which fixes the issues with the datetime sesnors.

**Comment by jazzyisj** [comment]
You guys are fast!  Installed and tested - thanks!  I'll close the issue.
